djstripe/stripe_objects.py

```diff
@@ -262,6 +303,57 @@ def _stripe_object_to_invoice(cls, target_cls, data):
         return target_cls._get_or_create_from_stripe_object(data, "invoice")[0]
 
     @classmethod
+    def _stripe_object_to_invoice_items(cls, target_cls, data, invoice):
+        """
+        Retrieves InvoiceItems for an invoice.
+
+        If the invoice item doesn't exist already then it is created.
+
+        If the invoice is an upcoming invoice that doesn't persist to the
+        database (i.e. ephemeral) then the invoice items are also not saved.
+
+        :param target_cls: The target class to instantiate per invoice item.
+        :type target_cls: ``StripeInvoiceItem``
+        :param data: The data dictionary received from the Stripe API.
+        :type data: dict
+        :param invoice: The invoice object that should hold the invoice items.
+        :type invoice: ``djstripe.models.Invoice``
+        """
+
+        lines = data.get("lines")
+        if not lines:
+            return []
+
+        invoiceitems = []
+        for line in lines.get("data", []):
+            if invoice.stripe_id:
+                save = True
+                line.setdefault("invoice", invoice.stripe_id)
+
+                if line.get("type") == "subscription":
+                    # Lines for subscriptions need to be keyed based on invoice and
+                    # subscription, because their id is *just* the subscription
+                    # when received from Stripe. This means that future updates to
+                    # a subscription will change previously saved invoices - Doing
+                    # the composite key avoids this.
+                    if not line["id"].startswith(invoice.stripe_id):
+                        line["id"] = "{invoice_id}-{subscription_id}".format(
+                            invoice_id=invoice.stripe_id,
+                            subscription_id=line["id"])
+            else:
+                # Don't save invoice items for ephemeral invoices
+                save = False
+
+            line.setdefault("customer", invoice.customer.stripe_id)
+            line.setdefault("date", int(dateformat.format(invoice.date, 'U')))
```

Comment:
Looks like that's [how it is designed](https://github.com/django/django/blob/31e0314979f6aa1c66828b55dd222c32aad94e9e/django/utils/dateformat.py#L35-L42) unfortunately - Although the function itself does cast to int explicitly, the wrapper for dateformat forces it back to a string.  It's slightly poor show that we go through int -> string -> int though.